File: python/src/wdl_model/model/expressions/wdl_float_literal.py
from __future__ import annotations

from .wdl_expression import ComponentType
from .wdl_number_literal import WdlNumberLiteral


class WdlFloatLiteral(WdlNumberLiteral[float]):
    def __init__(self, value: float | None = None):
        super().__init__(value)

    def negate(self) -> None:
        if self.value is not None:
            self.value = -self.value

    def componentType(self) -> ComponentType:
        return ComponentType.LITERAL

File: java/src/main/java/com/myriad/wdl/model/expressions/WdlFloatLiteral.java
package com.myriad.wdl.model.expressions;

public final class WdlFloatLiteral extends WdlNumberLiteral<Double> {
  public WdlFloatLiteral() {}

  public WdlFloatLiteral(double value) {
    super(value);
  }

  @Override
  public void negate() {
    if (this.value != null) {
      this.value = -this.value;
    }
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.FLOAT_LIT;
  }
}
